### PR TITLE
Added Support For Configurable Updatable Table Properties per Iceberg Catalog Type

### PR DESCRIPTION
**Pull request by srnand**
## Description
- For the use cases involving custom table properties per Iceberg catalog type, this update allows the specification of which properties are updatable for a given catalog. 
- Added the existing default behavior in the TrinoCatalog Iface so as to not break existing dependencies. 

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java [hunk 1]
```diff
@@ -196,4 +197,9 @@ void createMaterializedView(
     void updateColumnComment(ConnectorSession session, SchemaTableName schemaTableName, ColumnIdentity columnIdentity, Optional<String> comment);
 
     Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName, String hiveCatalogName);
+
+    default Set<String> getUpdatableTableProperties()
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java:201]: Is this an extension point for your fork?

Comment (srnand) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java:201]: It was a simple change that could be contributed back to OSS hence opened up for here.

Comment (srnand) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java:201]: @ebyhr and then I'll backport this change if this goes through. 
but even though I am a member of the trino community and I did the CLA thing, not sure why it says i am not authorized 🤔

Comment (ebyhr) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java:201]: First of all, I am not convinced with this change. We usually deny fork-specific changes. 

What do you mean > "i am not authorized"?

**Comment by ebyhr** [comment]
Let me close as a stale PR.

**Pull request closed without merging**